action.yml

```diff
@@ -36,19 +36,24 @@ inputs:
     description: 'If true, then the action will not expect the commits to have a verification signature. It is required to set this to true in GitHub Enterprise Server'
     required: false
     default: false
+  skip-verification:
+    type: boolean
+    description: 'If true, the action will not validate the user or the commit verification status'
+    default: false
 
 runs:
   using: 'composite'
   steps:
     - name: Fetch metadata
       id: dependabot-metadata
       uses: dependabot/fetch-metadata@v1
-      if: ${{ github.event_name == 'pull_request' && github.actor == 'dependabot[bot]' }}
+      if: github.event_name == 'pull_request' && (github.actor == 'dependabot[bot]' || skip-verification == 'true')
       with:
         skip-commit-verification: ${{ inputs.skip-commit-verification }}
+        skip-verification :  ${{ inputs.skip-verification }}
     - name: Merge/approve PR
       uses: actions/github-script@v6
-      if: ${{ github.event_name == 'pull_request' && github.actor == 'dependabot[bot]' }}
+      if: github.event_name == 'pull_request' && (github.actor == 'dependabot[bot]' || skip-verification == 'true')
```

Comment:
I'm not sure I understand this logic either. I mean, we want to execute the action regardless of whether skip-verification is true or false, don't we?

Reply:
I am not sure what you're trying to say. If `skip-verification` is the defaulted `false` value then the user should be validated. 

That is why it is an OR statement. We want to execute it if `github.actor` is `dependabot[bot]'` OR if `skip-verification` is set to `true`

That way, even non-dependabot users are also allowed to run this step as intended

Reply:
The logic is `(isDependabot || skipVerification) && isPullRequest`
So, it meets the target.

Reply:
you're right